controllers/openstackcluster_controller.go

```diff
@@ -196,24 +196,34 @@ func deleteBastion(log logr.Logger, osProviderClient *gophercloud.ProviderClient
 		return err
 	}
 
-	instance, err := computeService.GetInstanceByName(openStackCluster, fmt.Sprintf("%s-bastion", cluster.Name))
+	instanceStatus, err := computeService.GetInstanceByName(openStackCluster, fmt.Sprintf("%s-bastion", cluster.Name))
 	if err != nil {
 		return err
 	}
 
-	if instance != nil && instance.FloatingIP != "" {
-		if err = networkingService.DisassociateFloatingIP(openStackCluster, instance.FloatingIP); err != nil {
-			handleUpdateOSCError(openStackCluster, errors.Errorf("failed to disassociate floating IP: %v", err))
-			return errors.Errorf("failed to disassociate floating IP: %v", err)
+	instanceNS, err := instanceStatus.NetworkStatus()
+	if err != nil {
+		return err
+	}
+
+	if instanceStatus != nil {
+		floatingIP := instanceNS.FloatingIP()
+		if err != nil {
```

Comment:
You don't define `err` here, you can remove the err handling as no error will be returned

Reply:
Whoops. FloatingIP() returned an error in an earlier version and I missed this!